python/ray/worker.py

```diff
@@ -251,6 +254,10 @@ def __init__(self):
         # Identity of the driver that this worker is processing.
         self.task_driver_id = None
 
+        # Workers typically are not external clients
+        # TODO (dsuo): should rename this so it's clear what "client" means
```

Comment:
How about we use the terminology "external_client" instead of "client" throughout this PR. What do you think?

Reply:
A bit long, but it's more important to distinguish what ```client``` refers to. As long as you're ok with the following:

- ```EXTERNAL_CLIENT_MODE```
- ```external_client```
- ```ExternalClient```

I'm good!

Reply:
Yeah, descriptive is good.